modules/home-environment.nix

```diff
@@ -552,6 +552,9 @@ in
 
             cp ${activationScript} $out/activate
 
+            mkdir $out/bin
+            ln -s $out/activate $out/bin/home-manager-generation
```

Comment:
What is the usecase of symlinking to `$out/bin`?

Is it for `nix run`/`nix shell`?

Reply:
Yes, `home-manager switch --flake` uses `nix run <stuff>.activationPackage` internally ([here](https://github.com/ant-arctica/home-manager/blob/home-manager--flake/home-manager/home-manager#L222-L225)), like in my previous pr.

Reply:
Does `activationPackage` use `home-manager-generation` under the hood or should it be renamed to something else here?

Reply:
It's the same `activationPackage` attribute that get's built in the non flake version, if I understand it correctly it's the `home.activationPackage` option. But the name of the package is `home-manager-generation`, which is a bit confusing.